models/step_run_status.go

```diff
@@ -0,0 +1,88 @@
+package models
+
+// StepRunStatus ...
+type StepRunStatus int
+
+const (
+	StepRunStatusCodeSuccess StepRunStatus = iota
+	StepRunStatusCodeFailed
+	StepRunStatusCodeFailedSkippable
+	StepRunStatusCodeSkipped
+	StepRunStatusCodeSkippedWithRunIf
+	StepRunStatusCodePreparationFailed
+	StepRunStatusAbortedWithCustomTimeout   // step times out due to a custom timeout
+	StepRunStatusAbortedWithNoOutputTimeout // step times out due to no output received (hang)
+)
+
+func NewStepRunStatus(status string) StepRunStatus {
+	switch status {
+	case "success":
+		return StepRunStatusCodeSuccess
+	case "failed":
+		return StepRunStatusCodeFailed
+	case "failed_skippable":
+		return StepRunStatusCodeFailedSkippable
+	case "skipped":
+		return StepRunStatusCodeSkipped
+	case "skipped_with_run_if":
+		return StepRunStatusCodeSkippedWithRunIf
+	case "preparation_failed":
+		return StepRunStatusCodePreparationFailed
+	case "aborted_with_custom_timeout":
+		return StepRunStatusAbortedWithCustomTimeout
+	case "aborted_with_no_output":
+		return StepRunStatusAbortedWithNoOutputTimeout
+	default:
+		return -1
+	}
+}
+
+func (s StepRunStatus) String() string {
+	switch s {
+	case StepRunStatusCodeSuccess:
+		return "success"
+	case StepRunStatusCodeFailed:
+		return "failed"
+	case StepRunStatusCodeFailedSkippable:
+		return "failed_skippable"
+	case StepRunStatusCodeSkipped:
+		return "skipped"
+	case StepRunStatusCodeSkippedWithRunIf:
+		return "skipped_with_run_if"
+	case StepRunStatusCodePreparationFailed:
+		return "preparation_failed"
+	case StepRunStatusAbortedWithCustomTimeout:
+		return "aborted_with_custom_timeout"
+	case StepRunStatusAbortedWithNoOutputTimeout:
+		return "aborted_with_no_output"
+	default:
+		return "unknown"
+	}
+}
+
+func (s StepRunStatus) Name() string {
+	switch s {
+	case StepRunStatusCodeSuccess:
+		return "Success"
+	case StepRunStatusCodeFailed,
+		StepRunStatusCodePreparationFailed,
+		StepRunStatusCodeFailedSkippable,
+		StepRunStatusAbortedWithCustomTimeout,
+		StepRunStatusAbortedWithNoOutputTimeout:
+		return "Failed"
+	case StepRunStatusCodeSkipped,
+		StepRunStatusCodeSkippedWithRunIf:
+		return "Skipped"
+	default:
+		return "Unknown"
+	}
+}
+
+func (s StepRunStatus) IsFailure() bool {
+	if s == StepRunStatusCodeFailed ||
```

Comment:
Do we want to put all the failure statuses here. I originally removed them as a hack to match the step statuses doc.

Reply:
[bb6ce99](https://github.com/bitrise-io/bitrise/pull/836/commits/bb6ce9999b1d29192e255d29d2954d7472bbea4a)